### PR TITLE
#320 karma calculation and lock

### PR DESCRIPTION
**Pull request by patrickmohrmann**
the karma rituatl button is now refreshing current karma to maximum Karma

### CODE REVIEW

File: module/applications/actor/base-sheet.mjs [hunk 1]
```diff
@@ -120,4 +123,11 @@ export default class ActorSheetEd extends ActorSheet {
     const item = this.actor.items.get( li.dataset.itemId );
     return item.sheet?.render( true );
   }
+
+  async _onKarmaRefresh ( ) {
```
Comment (ChristopherSD) [inline module/applications/actor/base-sheet.mjs:127]: Just like with the weightCalculation, this should an appropriate method in the actor document, e.g. `this.actor.refreshKarma();`.
The function in the actor document class should like this, using the update method to persist the changes:
```javascript
karmaRitual() {
    this.update( {"system.karma.value": this.system.karma.max} );
}
```

**Review by ChristopherSD — CHANGES_REQUESTED** [review]
One change needed in base-sheet.mjs

File: module/applications/actor/base-sheet.mjs (hunk 1 above)
Comment (patrickmohrmann) [reply module/applications/actor/base-sheet.mjs:165]: ok, moved it as I understood it

Comment (ChristopherSD) [reply module/applications/actor/base-sheet.mjs]: Did some minor cleanup in commit 455b8f8cc3aee330be00d33c1a2ed8c1c9dc426a. If that is fine, mark this as resolved and we can merge it.

**Review by patrickmohrmann — COMMENTED** [review]
good

**Pull request merged**